stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/ServiceFile.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.net.grpc.builder.components;
+
+import org.ballerinalang.net.grpc.exception.BalGenerationException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Service file definition bean class.
+ */
+public class ServiceFile {
+    private boolean bidiStreaming;
+    private boolean clientStreaming;
+    private boolean unary;
+    private String serviceName;
+    private List<Method> unaryFunctions = new ArrayList<>();
+    private Method streamingFunction = null;
+
+    ServiceFile(String serviceName) {
```

Comment:
Can we make this constructor private as we need to use this only inside the builder class which is an inner class of this?